core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstreamreader/InputStreamReaderUnitTest.java

```diff
@@ -0,0 +1,37 @@
+package com.baeldung.inputstreamreader;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class InputStreamReaderUnitTest {
+    @Test
+    public void givenAStringWrittenToAFile_whenReadByInputStreamReader_thenShouldMatchWhenRead(@TempDir Path tempDir) throws IOException {
```

Comment:
This does not match the naming convention you had in the other class of tests

Can we choose a naming convention to settle on please?

Either:
1. givenAStringWrittenToAFile_whenReadByX_thenY
2. givenAString_whenWrittenToAFile_thenShouldYWhenReadByZ

I think option 1 matches better the purpose of this article, so let's rename all the InputStreamUnitTest methods to match this

Reply:
Renamed all unit test in InputStreamUnitTest to follow option 1.